### ISSUE TITLE
flex_attention does not output the full attention_weights with output_attention option

### ISSUE THREAD

**Issue by gblackout** [huggingface/transformers#36096]
### System Info

transformers==4.45.1
torch==2.6.0

### Information

- [x] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [x] My own task or dataset (give details below)

### Reproduction

```attention_interface: Callable = eager_attention_forward
        if self.config._attn_implementation != "eager":
            if self.config._attn_implementation == "sdpa" and kwargs.get("output_attentions", False):
                logger.warning_once(
                    "`torch.nn.functional.scaled_dot_product_attention` does not support `output_attentions=True`. Falling back to "
                    'eager attention. This warning can be removed using the argument `attn_implementation="eager"` when loading the model.'
                )
            else:
                attention_interface = ALL_ATTENTION_FUNCTIONS[self.config._attn_implementation]

        attn_output, attn_weights = attention_interface(
            self,
            query_states,
            key_states,
            value_states,
            attention_mask,
            dropout=0.0 if not self.training else self.attention_dropout,
            scaling=self.scaling,
            **kwargs,
        )

### Expected behavior

The output_attention option (e.g. in [llama](https://github.com/huggingface/transformers/blob/main/src/transformers/models/llama/modeling_llama.py#L232)) is supposed to return the full attention weights of shape (bsize, nheads, len_q, len_k). Currently, transformers support this option with eager attention and flex_attention, but it looks like what [flex_attention](https://github.com/huggingface/transformers/blob/main/src/transformers/integrations/flex_attention.py) returns are just the logsumexp scores of shape (bsize, nheads, len_q). It seems they do not support outputting the full attention weight either.

**Comment by gblackout** [comment]
The output_attention option (e.g. in [llama](https://github.com/huggingface/transformers/blob/main/src/transformers/models/llama/modeling_llama.py#L232)) is supposed to return the full attention weights of shape (bsize, nheads, len_q, len_k). Currently, transformers support this option with eager attention and flex_attention, but it looks like what [flex_attention](https://github.com/huggingface/transformers/blob/main/src/transformers/integrations/flex_attention.py) returns are just the logsumexp scores of shape (bsize, nheads, len_q). It seems they do not support outputting the full attention weight either.

**Comment by Rocketknight1** [comment]
I believe that flashattention and possibly flexattention don't support outputting weight scores because they aren't fully materialized during the computation, but I'm not certain about that! cc @ArthurZucker

**Comment by ArthurZucker** [comment]
I think this is related to the version of transformers no? We now fully pass the flag to `flex_attention` call that return the attention weights
